### PR TITLE
chore: Adding separate logic for showing signposting on query editor and code editor

### LINKED ISSUE #26689: Add separate logic for showing AI signposting on query editor and code editor

### PR DESCRIPTION
**Pull request by Druthi**
## Description
chore: Adding separate logic for showing signposting on query editor and code editor

#### PR fixes following issue(s)
Fixes #26689 

#### Type of change
- New feature (non-breaking change which adds functionality)

>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/utils/storage.ts [hunk 1]
```diff
@@ -444,9 +451,14 @@ export const setAIPromptTriggered = async () => {
   }
 };
 
-export const getAIPromptTriggered = async () => {
+export const getAIPromptTriggered = async (mode: string) => {
   try {
-    const flag: number | null = await store.getItem(STORAGE_KEYS.AI_TRIGGERED);
+    const storageKey =
+      mode === EditorModes.TEXT_WITH_BINDING
+        ? STORAGE_KEYS.AI_TRIGGERED
```
Comment (Rishabh-Rathod) [inline app/client/src/utils/storage.ts:458]: Can we add a more specific mode name instead of `AI_TRIGGERED`

**Review by Rishabh-Rathod — COMMENTED** [review]
small nit

File: app/client/src/utils/storage.ts (hunk 1 above)
Comment (Druthi) [reply app/client/src/utils/storage.ts:458]: Was already using this on production, hence didn't want to change it

Comment (Rishabh-Rathod) [reply app/client/src/utils/storage.ts:458]: yes, I meant only to change the key name to keep the functionality the same.

**Pull request merged**